### PR TITLE
Adding generator for products for firebase

### PR DESCRIPTION
**Pull request by devnook**
Hi Rowan, can you please take a look? I'm storing pies in firebase for now, but we can change it to other db of choice anytime.

### CODE REVIEW

**Comment by sgomes** [comment]
JS looks good. Deferring main review to someone with firebase experience.

File: src/piegen/firebase.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  databaseAuthVariableOverride: {
+    uid: 'my-service-worker',
+  },
+});
+
+const db = admin.database();
+const products = db.ref('/products');
+
+const pies = [];
+let i = 0;
```
Comment (samdutton) [inline src/piegen/firebase.js]: This is a bit nitpicky, but I'd be inclined just to use a simple `for` loop, something like this:

```
for (let i = 0;  i !== numPies; ++i) {
  pies.push(generatePie());
};
```

File: src/piegen/piegen.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  return keys[Math.floor(Math.random() * keys.length)];
+}
+
+function getRandomNumber(min, max) {
+  return Math.max(min, Math.round((Math.random() * max)));
+}
+
+const MIN_WEIGHT = 300; // In grams.
+const MAX_WEIGHT = 1000; // In grams.
+const MAX_PRICE = 800; // In grams.
```
Comment (samdutton) [inline src/piegen/piegen.js]: In grams :)?

File: src/piegen/piegen.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+const ALLERGEN_FREQUENCY = 0.7; // How often allergens are present.
+
+function generatePie() {
+  const pie = new Pie();
+  pie.topping = getRandomKey(ingredients.topping);
+  pie.price = getRandomNumber(0, MAX_PRICE);
+  pie.weight = getRandomNumber(MIN_WEIGHT, MAX_WEIGHT);
+  pie.calories = getRandomNumber(MIN_CALORIES, MAX_CALORIES);
+  if (Math.random() < ALLERGEN_FREQUENCY) {
+    let i = 0;
```
Comment (samdutton) [inline src/piegen/piegen.js]: This also might be simpler as a `for` loop.

**Comment by devnook** [comment]
PTAL

File: src/piegen/piegen.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
```
Comment (samdutton) [inline src/piegen/piegen.js]: Looks like this got left behind.

**Comment by samdutton** [comment]
LGTM aside from stray  `let i = 0`.

**Comment by devnook** [comment]
Thanks!

**Comment by samdutton** [comment]
LGTM

**Pull request merged**